Lib/unittest/test/testmock/testpatch.py

```diff
@@ -1854,5 +1855,31 @@ def foo(*a, x=0):
             self.assertEqual(foo(), 1)
         self.assertEqual(foo(), 0)
 
+    def test_patch_dict_with_orderdict(self):
+        foo = OrderedDict()
+        foo['a'] = object()
+        foo['b'] = 'python'
+
+        original = foo.copy()
+        update_values = list(zip('cdefghijklmnopqrstuvwxyz', range(26)))
+        patched_values = list(foo.items()) + update_values
+
+        with patch.dict(foo, OrderedDict(update_values)):
+            self.assertEqual(list(foo.items()), patched_values)
+
+        self.assertEqual(foo, original)
+
+        with patch.dict(foo, update_values):
+            self.assertEqual(list(foo.items()), patched_values)
+
+        self.assertEqual(foo, original)
+
+        # check if removing foo element not fail
```

Comment:
I don't get why this was added. Any specific reason why this is needed in this test? Is there a case where this can fail perhaps due to `OrderedDict`?

Reply:
I am just testing if the I modify the foo dict, this don't have to be modified out of the context

Reply:
Yes, I think it's not related to the the test in question that is order of OrderedDict in patch.dict and more about a general one in whether being able to pop an element from a patched dict.

Reply:
Ready